include/zephyr/spinlock.h

```diff
@@ -199,16 +233,47 @@ static ALWAYS_INLINE int k_spin_trylock(struct k_spinlock *l, k_spinlock_key_t *
 
 	z_spinlock_validate_pre(l);
 #ifdef CONFIG_SMP
+#ifdef CONFIG_TICKET_SPINLOCKS
```

Comment:
can this become `#if defined(CONFIG_SMP) && defined(CONFIG_TICKET_SPINLOCKS)` instead?

Reply:
the pattern is also in other places in this PR

Reply:
Hi @nashif , 

CONFIG_SMP is not related exclusively to ticket spinlocks.
Both the original and the ticket spinlock implementations come under CONFIG_SMP.
I tried not to alter the original spinlock implementation in any way, so I used the same pattern everywhere:

```
#ifdef CONFIG_TICKET_SPINLOCKS
// ticket implementation
#else
// original implementation
#endif
```